### PR TITLE
set adamw_mode default true (follows FusedAdam and < 0.3.11 logic)

### PR DESCRIPTION
**Pull request by jeffra**
Was going to also update our docs related to this but noticed they are currently indicating this behavior anyway.

image

The user can explicitly set adam_w_mode=false if they truly want pure adam logic.

### CODE REVIEW

**Review by conglongli — APPROVED** [review]
LGTM

File: deepspeed/runtime/engine.py [hunk 1]
```diff
@@ -640,10 +640,10 @@ def _configure_basic_optimizer(self, model_parameters):
 
         if self.optimizer_name() in [ADAM_OPTIMIZER, ADAMW_OPTIMIZER]:
             torch_adam = optimizer_parameters.pop(TORCH_ADAM_PARAM, False)
-            adam_w_mode = self.optimizer_name() == ADAMW_OPTIMIZER
+            adam_w_mode = optimizer_parameters.get(ADAM_W_MODE, ADAM_W_MODE_DEFAULT)
```
Comment (samyam) [inline deepspeed/runtime/engine.py:643]: I am a bit confused on why we have ADAMW_OPTIMIZER and ADAM_W_MODE both. What if the user passes adamw optimizer and sets adam_w_mode to false?

Comment (samyam) [reply deepspeed/runtime/engine.py:643]: Does it make sense to get rid of ADAMW_OPTIMIZER, and set ADAM_W_MODE to True by default?

Comment (conglongli) [reply deepspeed/runtime/engine.py:643]: @samyam We added AdamW optimizer type last month because of https://github.com/microsoft/DeepSpeed/issues/668

Comment (conglongli) [reply deepspeed/runtime/engine.py:643]: > What if the user passes adamw optimizer and sets adam_w_mode to false?

I do agree with @samyam about this. @jeffra maybe we want to assert adam_w_mode as true when ADAMW_OPTIMIZER is used?

Comment (jeffra) [reply deepspeed/runtime/engine.py]: resolved offline with new structure

**Review by samyam — APPROVED** [review]

**Review by ShadenSmith — APPROVED** [review]

**Pull request merged**